File: BehaviorPattern/Strategy/Python/Contexts/ShoppingCart.py

class ShoppingCart:
    """
        Context class Shopping Cart
    """
    def __init__(self, *args, **kargs):
        self.paymentStrategy = kargs.get('paymentStrategy', None)
        if args:
            self.paymentStrategy = args[0]
            
    def setPaymentStrategy(self, paymentStrategy):
        self.paymentStrategy = paymentStrategy
        
    def checkout(self, amount):
        self.paymentStrategy.pay(amount)
        

File: BehaviorPattern/Strategy/Golang/Contexts/ShoppingCart.go

package contexts

import (
	iface "v/Interfaces"
)

// Context Class
type ShoppingCart struct {
	paymentStrategy iface.PaymentStrategy
}

// Constructor 
func NewShoppingCart(paymentStrategy iface.PaymentStrategy) *ShoppingCart {
	return &ShoppingCart{paymentStrategy: paymentStrategy}
}

func (c *ShoppingCart) SetPaymentStrategy(paymentStrategy iface.PaymentStrategy) {
	c.paymentStrategy = paymentStrategy
}

func (c *ShoppingCart) Checkout(amount int) {
	c.paymentStrategy.Pay(amount);
}